src/media/image/imagemedia.service.ts

```diff
@@ -0,0 +1,121 @@
+import { createReadStream, ReadStream } from 'fs';
+import { rm, writeFile } from 'fs/promises';
+import { Injectable } from '@nestjs/common';
+import { ImageMedia, ImageMediaPreset } from '@prisma/client';
+import { ConfigModule } from '../../config/config.module';
+import { PrismaService } from '../../prisma/prisma.service';
+import { MulterWithMime } from '../../upload.interceptor';
+import { User } from '../../users/interfaces/user.interface';
+import ImageMediaUploadReqDto from './dto/req/imagemedia-upload-req.dto';
+import sharp from 'sharp';
+
+export type ConversionOptions = Omit<ImageMediaUploadReqDto, 'public'>;
+
+type PresetStruct = Record<
+  Exclude<ImageMediaPreset, typeof ImageMediaPreset.CUSTOM>,
+  Omit<ConversionOptions, 'preset'>
+>;
+const presets: PresetStruct = {
+  AVATAR: { width: 256, height: 256, quality: 70, effort: 5 },
+};
+
+export type ImageMetadata = Omit<ImageMedia, 'id' | 'uploadedAt' | 'isPublic' | 'uploaderId'>;
+
+@Injectable()
+export class ImageMediaService {
+  constructor(
+    readonly prisma: PrismaService,
+    readonly config: ConfigModule,
+  ) {}
+
+  async convertMedia(file: MulterWithMime, options: ConversionOptions): Promise<ImageMetadata> {
+    if (!(options.preset in presets)) options.preset = ImageMediaPreset.CUSTOM;
+    if (options.preset) Object.assign(options, presets[options.preset]);
```

Comment:
Je crois que tu peux enlever le `if`, si c'était `null`, `!(options.preset in presets)` serait `true`, tu serais passé dans la condition au dessus